server/server.c

```diff
@@ -157,13 +161,39 @@ static int process_active_lobby(Lobby* lobby, int id) {
     msg.id = GAME_STATE_UPDATE;
     msg.game_state_update.state = lobby->game.state;
 
-    if (send_message(lobby->guest, &msg) < 0) {
+
+    if (send_message(lobby->owner, &msg) < 0) {
       return -1;
     }
 
-    if (send_message(lobby->owner, &msg) < 0) {
+    msg.game_state_update.state = lobby->game.state == STATE_LOST ? STATE_WON : STATE_LOST;
+    if (send_message(lobby->guest, &msg) < 0) {
       return -1;
     }
+
+    return 0;
+  }
+
+  ServerMessage response;
+
+  response.id = SERVER_UPDATE;
+
+  response.server_update.ball_position.x = lobby->game.ball.position.x;
+  response.server_update.ball_position.y = -1 * lobby->game.ball.position.y;
+
+  response.server_update.opponent_position.x = lobby->game.player.position.x;
+  response.server_update.opponent_position.y = -1 * lobby->game.player.position.y;
+
+  if (send_message(lobby->guest, &response) < 0) {
+    return -1;
+  }
+
+  response.server_update.ball_position.y = lobby->game.ball.position.y;
+  response.server_update.opponent_position.x = lobby->game.opponent.position.x;
+  response.server_update.opponent_position.y = -1 * lobby->game.opponent.position.y;
```

Comment:
This is incorrect. Opponent position should not be mirrored for owner of the lobby.